core/src/language/go.rs

```diff
@@ -26,38 +28,43 @@ pub struct Go {
 }
 
 impl Language for Go {
-    fn generate_types(&mut self, w: &mut dyn Write, data: &ParsedData) -> std::io::Result<()> {
+    fn generate_types(
+        &mut self,
+        w: &mut dyn Write,
+        _imports: &CrateTypes,
+        data: ParsedData,
+    ) -> std::io::Result<()> {
         // Generate a list of all types that either are a struct or are aliased to a struct.
         // This is used to determine whether a type should be defined as a pointer or not.
-        let mut types_mapping_to_struct = HashSet::new();
-        for s in &data.structs {
-            types_mapping_to_struct.insert(s.id.original.as_str());
-        }
+        let mut types_mapping_to_struct =
+            HashSet::from_iter(data.structs.iter().map(|s| s.id.original.clone()));
+
         for alias in &data.aliases {
-            if types_mapping_to_struct.contains(&alias.r#type.id()) {
-                types_mapping_to_struct.insert(alias.id.original.as_str());
+            if types_mapping_to_struct.contains(alias.r#type.id()) {
+                types_mapping_to_struct.insert(alias.id.original.clone());
             }
         }
 
-        self.begin_file(w)?;
+        self.begin_file(w, &data)?;
 
-        let mut items: Vec<RustItem> = vec![];
+        let ParsedData {
+            structs,
+            enums,
+            aliases,
+            ..
+        } = data;
 
-        for a in &data.aliases {
-            items.push(RustItem::Alias(a.clone()))
-        }
-
-        for s in &data.structs {
-            items.push(RustItem::Struct(s.clone()))
-        }
-
-        for e in &data.enums {
-            items.push(RustItem::Enum(e.clone()))
-        }
+        let mut items = Vec::from_iter(
+            aliases
+                .into_iter()
+                .map(RustItem::Alias)
+                .chain(structs.into_iter().map(RustItem::Struct))
+                .chain(enums.into_iter().map(RustItem::Enum)),
+        );
```

Comment:
minor: Use `.collect` instead of `from_iter`.

Reply:
Updated